cli/src/template.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use splinter::circuit::template::{
+    Builders, CircuitCreateTemplate, CircuitTemplateError, CircuitTemplateManager, RuleArgument,
+};
+
+use crate::error::CliError;
+
+const NODES_ARG: &str = "nodes";
+
+pub struct CircuitTemplate {
+    template: CircuitCreateTemplate,
+    arguments: HashMap<String, String>,
+}
+
+impl CircuitTemplate {
+    pub fn list_available_templates() -> Result<Vec<String>, CliError> {
+        let manager = CircuitTemplateManager::default();
+        let templates = manager.list_available_templates()?;
+        Ok(templates)
+    }
+
+    pub fn load_raw(name: &str) -> Result<String, CliError> {
```

Comment:
Can you add comments here explaining the difference between load_raw and load?

Reply:
added

Reply:
I dont see the comments? Did you add them to the CircuitTemplateManager instead?

Reply:
yes, my bad. Just added comments here too